catalog/imaging/imaging-plugin-nitf/src/main/java/org/codice/alliance/plugin/nitf/NitfRenderPlugin.java

```diff
@@ -0,0 +1,386 @@
+ */
+public class NitfRenderPlugin implements PostProcessPlugin {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(NitfRenderPlugin.class);
+
+    private static final String NULL_INPUT_MSG = "Unable to process null input.";
+
+    private static final String IMAGE_NITF = "image/nitf";
+
+    static final MimeType NITF_MIME_TYPE;
+
+    static {
+        try {
+            NITF_MIME_TYPE = new MimeType(IMAGE_NITF);
+        } catch (MimeTypeParseException e) {
+            throw new ExceptionInInitializerError(String.format(
+                    "Unable to create MimeType from '%s': %s",
+                    IMAGE_NITF,
+                    e.getMessage()));
+        }
+    }
+
+    private static final String JPG = "jpg";
+
+    private static final int DEFAULT_MAX_NITF_SIZE_MB = 120;
+
+    private int maxNitfSizeMB = DEFAULT_MAX_NITF_SIZE_MB;
+
+    private boolean storeOriginalImage = true;
+
+    private boolean createOverview = true;
+
+    private static final double DEFAULT_MAX_SIDE_LENGTH = 1024.0;
+
+    private double maxSideLength = DEFAULT_MAX_SIDE_LENGTH;
+
+    static {
+        IIORegistry.getDefaultInstance()
+                .registerServiceProvider(new J2KImageReaderSpi());
+    }
+
+    @Override
+    public ProcessRequest<ProcessCreateItem> processCreate(
+            ProcessRequest<ProcessCreateItem> processRequest) throws PluginExecutionException {
+        return processRequest(processRequest);
+    }
+
+    @Override
+    public ProcessRequest<ProcessUpdateItem> processUpdate(
+            ProcessRequest<ProcessUpdateItem> processRequest) throws PluginExecutionException {
+        return processRequest(processRequest);
+    }
+
+    @Override
+    public ProcessRequest<ProcessDeleteItem> processDelete(
+            ProcessRequest<ProcessDeleteItem> processRequest) throws PluginExecutionException {
+        if (processRequest == null) {
+            throw new PluginExecutionException(NULL_INPUT_MSG);
+        }
+
+        //No work to do
+        return processRequest;
+    }
+
+    private <T extends ProcessResourceItem> ProcessRequest<T> processRequest(
+            ProcessRequest<T> input) throws PluginExecutionException {
+        if (input == null) {
+            throw new PluginExecutionException(NULL_INPUT_MSG);
+        }
+
+        List<T> processResourceItems = input.getProcessItems();
+
+        if (processResourceItems != null) {
+            List<T> newProcessResourceItems = new ArrayList<>();
+
+            for (T processResourceItem : processResourceItems) {
+                ProcessResource processResource = processResourceItem.getProcessResource();
+                Metacard metacard = processResourceItem.getMetacard();
+
+                if (!isNitfMimeType(processResource.getMimeType())) {
+                    LOGGER.trace("Skipping process item: name={} mimeType={}.",
+                            processResource.getName(),
+                            processResource.getMimeType());
```

Comment:
In the case that we have a large set of items being ingested, this may overwhelm the logs.  If it's true that non-NITFs are passed to this plugin, I don't see why this logging statement is beneficial because it's only going to tell us that a file was skipped (because it's not a NITF).